reload/fs.go

```diff
@@ -0,0 +1,60 @@
+package reload
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+)
+
+type FS struct {
+	embed fs.FS
+	dir   fs.FS
+}
+
+func NewFS(embed fs.ReadDirFS, dir string) FS {
+	return FS{
+		embed: embed,
+		dir:   os.DirFS(dir),
+	}
+}
+
+func (f FS) Open(name string) (fs.File, error) {
+	if name == "embed.go" {
+		return nil, fs.ErrNotExist
+	}
+	file, err := f.getFile(name)
+	if name != "." {
+		return file, err
+	}
+	return rootFile{file}, err
+}
+
+func (f FS) getFile(name string) (fs.File, error) {
+	cfgFile := "./.buffalo.dev.yml"
+	if _, err := os.Stat(cfgFile); err != nil {
+		return f.embed.Open(name)
+	}
+	return f.dir.Open(name)
+}
+
+type rootFile struct {
+	fs.File
+}
+
+func (f rootFile) ReadDir(n int) ([]fs.DirEntry, error) {
+	dir, ok := f.File.(fs.ReadDirFile)
+	if !ok {
+		return nil, fmt.Errorf("failed at hiding embed.go file")
```

Comment:
Is it a correct error message? It could be "called on incompatible object" or something.

Reply:
I replaced it with an error that `f` isn't a directory.